### PR TITLE
feat(reporter): customers can now catch every errors thrown by forest

### PR DESCRIPTION
**Pull request by SteveBunlon**
### General

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Test manually the implemented changes
- [x] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [x] Consider the security impact of the changes made

Linked PR https://github.com/ForestAdmin/forest-rails/pull/325

Clickup card https://app.clickup.com/t/9r265k

### CODE REVIEW

**Review by jeffladiray — APPROVED** [review]
LGTM. 
As discussed, I feel like a couple of others FOREST_LOGGER.error could have been also handled by your FOREST_REPORTER. However, it does match with #325, so it's ok for me

File: spec/config/initializers/logger_spec.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module ForestLiana
   describe Logger do
-    describe 'self.log' do
+    describe 'logging'
```
Comment (arnaudbesnier) [inline spec/config/initializers/logger_spec.rb:3]:
```suggestion
    describe 'logging' do
```
?

File: config/initializers/logger.rb [hunk 2]
```diff
@@ -1,29 +1,56 @@
...
+    end
+
+    def debug(message)
+      @logger.debug message
     end
   end
 end
 
-FOREST_LOGGER = ForestLiana::Logger.log
+FOREST_LOGGER = ForestLiana::Logger.new
```
Comment (arnaudbesnier) [inline config/initializers/logger.rb:56]: [Style] missing new line

File: config/initializers/logger.rb [hunk 3]
```diff
@@ -1,29 +1,56 @@
...
+        @reporter.report error
+      end
+    end
+
+    def info(message)
+      @logger.info message
+    end
+
+    def debug(message)
+      @logger.debug message
```
Comment (arnaudbesnier) [inline config/initializers/logger.rb:51]: Not sure why all these methods must be redefined manually.

File: app/controllers/forest_liana/actions_controller.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ def get_smart_action_hook_request
       begin
         params[:data][:attributes]
       rescue => error
-        FOREST_LOGGER.error "Smart Action hook request error: #{error}"
+        FOREST_LOGGER.error "Smart Action hook request error: #{error}", error
```
Comment (arnaudbesnier) [inline app/controllers/forest_liana/actions_controller.rb:8]: From my point of view `FOREST_LOGGER` purpose is to print logs and only do so.

I would not feel it strange to read:
```suggestion
        FOREST_LOGGER.error "Smart Action hook request error: #{error}"
        FOREST_REPORTER.report error
```

**Review by arnaudbesnier — COMMENTED** [review]
I think I do prefer to de-couple logging and reporting like it was done in the initial Pull Request.

**Comment by SteveBunlon** [comment]
Thanks @jeffladiray and @arnaudbesnier for the review. 

I do listen your POV and I decided to rollback to previous version where LOGGER and REPORTER are decoupled. Let's continue that way 💪

**Review by jeffladiray — APPROVED** [review]

**Pull request merged**